File: quant/api/schemas/strategies.py
"""Pydantic schemas for ``GET /api/v1/strategies`` — Phase 1.6."""

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class StrategyListRow(BaseModel):
    """One deployable BT.STRATEGY version for the Trade picker (caller-owned)."""

    strategy_id: UUID
    strategy_vid: int
    strategy_nm: str | None = None
    is_best_ind: str
    created_at: datetime
    sharpe_ratio: float | None = None
    calmar_ratio: float | None = None
    max_drawdown: float | None = None
    total_return: float | None = None
    annualized_return: float | None = None

File: frontend/src/types/strategies.ts
/** Mirrors quant/api/schemas/strategies.py — keep in sync. */

export interface StrategyListRow {
  strategy_id: string;
  strategy_vid: number;
  strategy_nm: string | null;
  is_best_ind: string;
  created_at: string;
  sharpe_ratio: number | null;
  calmar_ratio: number | null;
  max_drawdown: number | null;
  total_return: number | null;
  annualized_return: number | null;
}

export type StrategyListVersions = 'best' | 'all';

/** Optional react-router state when navigating to Trade Apply. */
export interface TradeApplyLocationState {
  strategyId?: string;
  strategyVid?: number;
  strategyNm?: string | null;
  queueId?: string;
}
